src/Tizen.Network.Nsd/Tizen.Network.Nsd/DnssdService.cs

```diff
@@ -217,15 +219,78 @@ public IPAddressInformation IP
                 return IPAddressInstance;
             }
         }
-
-        private void GetTxtRecord(out ushort length, out byte[] value)
+        
+        /// <summary>
+        /// Returns TXT records in the form of key and value.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 9 </since_tizen>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public IDictionary<string, string> GetTxtRecord()
+        {
+            IntPtr data;
+            ushort length;
+            var map = new Dictionary <string, string>();
+            
+            int ret = Interop.Nsd.Dnssd.GetAllTxtRecord(_serviceHandle, out length, out data);
+            if (ret != (int)DnssdError.None)
+            {
+                Log.Error(Globals.LogTag, "Failed to get the TXT record, Error: " + (DnssdError)ret);
+                NsdErrorFactory.ThrowDnssdException(ret);
+            }
+            if (length > 0)
+            {
+                byte[] managedArray = new byte[length];
+                Marshal.Copy(data, managedArray, 0, length);
+                var index = 0;
+                while (index < length)
+                {
+                    var txtLen = managedArray[index++];
+                    if (txtLen > length)
+                        Log.Debug(Globals.LogTag, "Invalid data");
+                    
+                    string record = Encoding.UTF8.GetString(managedArray, index, txtLen);
+                    string[] list = record.Split('=');
+                    map.Add(list[0], list[1]);
+                    index += txtLen;
+                }
+            }
+            return map;
+        }
+        
+        /// <summary>
+        /// Gets all raw TXT records.
+        /// </summary>
+        /// <remarks>
+        /// After adding the TXT record call for get TXT record.
+        /// </remarks>
+        /// <since_tizen> 4 </since_tizen>
+        /// <param name="value">The value of the TXT record. If txt record not availabe added with an empty value.</param>
+        /// <feature>http://tizen.org/feature/network.service_discovery.dnssd</feature>
+        /// <exception cref="NotSupportedException">Thrown when DNS-SD is not supported.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when any other error occurred.</exception>
+        public void GetTxtRecord(out byte[] value)
```

Comment:
How about just returning byte array instead of providing out parameter?
```suggestion
        public byte[] GetTxtRecordAsBytes()
```
And this method can be used in `public IDictionary<string, string> GetTxtRecord()` method also.

Reply:
And why should this method be made public?

Reply:
Application need api to take decision on txt record data
Applied review comments